index.js

```diff
@@ -33,10 +34,12 @@ module.exports = function(content) {
       " + ': ' + exception); };"
     );
   } else {
+    const filePathArray = (defaultConfig.filenameBasePath || []).concat(fileName);
+    const filePath = JSON.stringify(filePathArray).slice(1, -1); // to remove '[' and ']'
```

Comment:
If the `filenameBasePath` is set to `["foo", "bar", "app.asar"]` and the `fileName` is `"file.node"`, this will result in:

```js
> const filePathArray = (["foo", "bar", "app.asar"] || []).concat("file.node")
undefined
> const filePath = JSON.stringify(filePathArray).slice(1, -1);
undefined
> path.resolve(".", filePath)
'/Users/smefju/tmp/"foo","bar","app.asar","file.node"'
> filePath
'"foo","bar","app.asar","file.node"'
>
```

We need to join the array given by concatenating `filePathArray` and `[filePath]` using a proper character.

I think that you can just make it a string instead of an array: `filenameBasePath: "base/path/to/package"`

Reply:
@smt116 We need to generate code so I think your test code is not correct. I would recommend you to use `eval` to test my code as follows.

```
> const filePathArray = (["foo", "bar", "app.asar"] || []).concat("file.node")
undefined
> const filePath = JSON.stringify(filePathArray).slice(1, -1);
undefined
> eval('path.resolve(".", ' + filePath + ')')
'C:\\Users\\XXXX\\foo\\bar\\app.asar\\file.node'
```

I believe `filePath` should be passed to `path.resolve` as a string literal and we should avoid embedding a path delimiter like `/` because it is platform-dependent so should keep an array instead of a string.

Reply:
You are right ;)